lib/defra_ruby/exporters/registration_export_service.rb

```diff
@@ -50,19 +76,38 @@ def save_to_s3(export_type, file_path)
         result
       end
 
-      private
+      def self.presigned_url(export_type, file_name)
+        export_bucket(export_type)
+          .object(file_name)
+          .presigned_url(
+            :get,
+            expires_in: 20 * 60, # 20 minutes in seconds
+            secure: true,
+            response_content_type: "text/csv",
+            response_content_disposition: "attachment; filename=#{file_name}"
+          )
+      end
+
+      private_class_method def self.config
+        DefraRuby::Exporters.configuration
+      end
+
+      private_class_method def self.date_range_helper
+        DefraRuby::Exporters::Helpers::DateRange
+      end
 
-      def full_path(file_name)
+      private_class_method def self.full_path(file_name)
         Rails.root.join "tmp", file_name
       end
 
-      def export_bucket(export_type)
+      private_class_method def self.export_bucket(export_type)
         aws_config = {
-          epr: { credentials: EPR_EXPORT_AWS_CREDENTIALS, bucket: EPR_EXPORT_S3_BUCKET }
+          epr: { credentials: config.epr_export_aws_credentials, bucket: config.epr_export_s3_bucket },
```

Comment:
I'm of the opinion now there there is more work being done here, and we now have a bonafide config class that the generation of the credentials hash could be moved to it. So the content of this method becomes something like

```ruby
        aws_config = config.aws_config(export_type)
        s3 = Aws::S3::Resource.new(
          region: aws_config[:region],
          credentials: aws_config[:credentials]
        )

        s3.bucket(aws_config[:bucket])
```

Reply:
Fixed in 078a31f.